ComfySoda_Unity2018.2.7f1/Assets/Scripts/ComySodaEngine/GameOverController.cs

```diff
@@ -40,6 +40,7 @@ private void OnDestroy()
         public static void StartGameOver()
         {
             function();

```

Comment:
This is the stupidest singleton I have ever seen. Declare a proper static instance. Or make private static events that your instance subscribes to and your static methods invoke. But this is just stupid. I know SDSU has a professor that teaches this in C++. It is bad practice in C++. It is worse in C#.

Reply:
The following is a proper static instance implementation that will do what you want to do without bad hacks. I will leave it up to you to explore the private static events - instance subscriber approach if you choose to do so.
``` csharp
private static GameOverController gameOverController;
...
public static void StartGameOver()
{
    gameOverController.StartCoroutine(GameOverDisplay());
}
```